pkgs/tools/package-management/nix/common.nix

```diff
@@ -157,7 +160,7 @@ stdenv.mkDerivation {
   ] ++ lib.optionals (!atLeast24) [
     # option was removed in 2.4
     "--disable-init-state"
-  ] ++ lib.optionals stdenv.isLinux [
+  ] ++ lib.optionals (stdenv.isLinux && !enableStatic) [
```

Comment:
not having a sandbox shell leads to very hard to debug build failures in my experience so it's a bit sad. Could we copy the shell to $out and somehow make a relative reference ?

Reply:
I think the goal was to have only a single executable that can be deployed. And you point to issues that can be caused by this? Otherwise I think there might be different usecases of static builds and we could make it configurable.

Reply:
Ideally, the sandbox shell would be part of the nix binary itself. Or be part of the derivation closure as a dependency. I think there is more work to be done down the line here.

Another aspect is that if a project doesn't build, as long as /bin/sh is POSIX compatible, this is really a bug of the project itself.

Reply:
> as long as /bin/sh is POSIX compatible, this is really a bug of the project itself.

I think it must be statically linked (the shared libraries it depends on are not brought into the sandbox)

Reply:
https://github.com/NixOS/nixpkgs/pull/212304